cpp/include/cudf/column/column_view.hpp

```diff
@@ -473,4 +474,47 @@ class mutable_column_view : public detail::column_view_base {
  *---------------------------------------------------------------------------**/
 size_type count_descendants(column_view parent);
 
-}  // namespace cudf
\ No newline at end of file
+namespace detail {
+/**---------------------------------------------------------------------------*
+ * @brief Constructs a zero-copy `column_view`/`mutable_column_view` of the
+ * elements in the range `[begin,end)` in `input`.
+ *
+ * @note It is the caller's responsibility to ensure that the returned view
+ * does not outlive the viewed device memory.
+ *
+ * @throws `cudf::logic_error` if `begin` with `end` goes beyond
```

Comment:
Do you mean `begin` *or* `end`? "with" doesn't make much sense here.

Maybe replace this all with a single line:

```suggestion
 * @throws `cudf::logic_error` if `begin < 0`, `end < begin` or `end > input.size()`.
```